channeldb/messagestore_test.go

```diff
@@ -0,0 +1,242 @@
+	// We'll construct the expected response. Only the first two peers will
+	// have messages.
+	totalPeerMsgs := make(map[[33]byte]map[uint64]lnwire.MessageType, 2)
+	for i := 0; i < 2; i++ {
+		totalPeerMsgs[peers[i]] = expectedPeerMsgs[i]
+	}
+
+	msgs, err := msgStore.Messages()
+	if err != nil {
+		t.Fatalf("unable to retrieve all peers with pending messages: "+
+			"%v", err)
+	}
+	if len(msgs) != len(totalPeerMsgs) {
+		t.Fatalf("expected %d peers with messages, got %d",
+			len(totalPeerMsgs), len(msgs))
+	}
+	for peer, peerMsgs := range msgs {
+		expected, ok := totalPeerMsgs[peer]
+		if !ok {
+			t.Fatalf("expected to find pending messages for peer %x",
+				peer)
+		}
+
+		assertPeerMsgs(peerMsgs, expected)
+	}
+}
+
+// TestGossipMessageStoreUnsupportedMessage ensures that we are not able to add
+// a message which is unsupported.
+func TestGossipMessageStoreUnsupportedMessage(t *testing.T) {
+	t.Parallel()
+
+	// We'll start by creating our test message store.
+	msgStore, cleanUp := createTestMessageStore(t)
+	defer cleanUp()
+
+	// Create a message that is known to not be supported by the store.
+	peer := randPubKey(t)
+	unsupportedMsg := &lnwire.NodeAnnouncement{}
+
+	// Attempting to add it to the store should result in
+	// ErrUnsupportedMessage.
+	err := msgStore.AddMessage(unsupportedMsg, peer)
+	if err != ErrUnsupportedMessage {
+		t.Fatalf("expected ErrUnsupportedMessage, got %v", err)
+	}
+}
+
+// TestLegacyGossipMessageStore ensures a message with a legacy key format
+// within the store can be properly removed from it.
+func TestLegacyGossipMessageStore(t *testing.T) {
+	t.Parallel()
+
+	// We'll start by creating our test message store.
+	msgStore, cleanUp := createTestMessageStore(t)
+	defer cleanUp()
+
+	// Create a message which we'll insert into the store with a legacy key.
+	peerPubKey := randPubKey(t)
+	msg := randAnnounceSignatures(t)
+	legacyMsgKey := oldMessageStoreKey(msg, peerPubKey)
+	err := msgStore.db.Update(func(tx *bbolt.Tx) error {
+		messageStore := tx.Bucket(messageStoreBucket)
+		var b bytes.Buffer
+		if _, err := lnwire.WriteMessage(&b, msg, 0); err != nil {
+			return err
+		}
+		return messageStore.Put(legacyMsgKey, b.Bytes())
+	})
+	if err != nil {
+		t.Fatalf("unable to add message with legacy key: %v", err)
+	}
+
+	// Then, we'll ensure we can properly retrieve it.
+	msgs, err := msgStore.MessagesForPeer(peerPubKey)
+	if err != nil {
+		t.Fatalf("unable to retrieve messages for peer: %v", err)
+	}
+	if len(msgs) != 1 {
+		t.Fatalf("expected 1 message for peer, got %d", len(msgs))
+	}
+
+	// Finally, ensure we can properly remove the message from the store.
+	if err := msgStore.DeleteMessage(msg, peerPubKey); err != nil {
+		t.Fatalf("unable to delete legacy message: %v", err)
+	}
+	msgs, err = msgStore.MessagesForPeer(peerPubKey)
+	if err != nil {
+		t.Fatalf("unable to retrieve messages for peer: %v", err)
+	}
+	if len(msgs) != 0 {
+		t.Fatalf("expected 0 messages for peer, got %d", len(msgs))
+	}
```

Comment:
Maybe add with a new key, ensure it overwrites the legacy one?

Reply:
The current implementation doesn't check for a legacy entry when adding the same message again. Are you saying that should be done and the test should also be expanded to cover this? In any case, don't think this is really necessary as we'll end up removing the message (both new and legacy entries) once it's stale anyway.